### DISCUSSION TITLE
How to remove code programmatically

### DISCUSSION THREAD

**Discussion by ronaldohoch** [GrapesJS/grapesjs#3960] [Q&A] [answered]
Hello Artf! 

I have these use cases and i think i need the same solution in both cases.

_The first is:_ When user click on Preview button, i need to add some javascripts to the page, so he can test a form integration on it, and when he goes back, i want to remove the javascript.
_Second case:_ I need to create a panel that user will add the API Keys from GTM, Facebook Pixel, Google API  Key and Google Fonts, in these case, i need to add in any moment, but i need the hability to remove it too.
All because, i need to save on the DB only the HTML and CSS of page.

Here the live Demo: https://jsfiddle.net/ar3s2mwh/

Edit: Also, my form component is working as it is described in Component & Js session of documentation.

Some prints:

Here i'm quering the element by myself and removing it.
![image](https://user-images.githubusercontent.com/2287371/142421071-349078aa-ab99-4b8a-a926-624f9b80ba4d.png)

Here you can see that it doesn't exist in the document anymore.
![image](https://user-images.githubusercontent.com/2287371/142421239-d7c9c51b-b2bd-429a-a1f9-28c82263b28e.png)

But when i call `editor.getHtml()`, the script is there @_@
![image](https://user-images.githubusercontent.com/2287371/142421464-e746dc64-fada-446c-afc0-2be7b7a50aed.png)

And here, when i check the export modal.
![image](https://user-images.githubusercontent.com/2287371/142421568-1d4c6ca0-5ceb-47d8-b47e-8219bcc550ba.png)

So, i didn't find a way to safe remove an element from canvas and it update to the original returned file

**Comment by gustavohleal** [comment, marked as answer]
As we work together, we solved this problem by looking at API reference rsrs

I noticed that `editor.Canvas` gets the canvas itself as a html element of the page and manage him, not the elements dropped inside him.

The solution was use `editor.DomComponents` that actually works with what is inside the canvas. That means the elements we are inserting to edit and create a page or something.

So using this way:

```javascript
  const body =  editor.DomComponents.getElements(); // Returns wrapper's children collection
  // Once we have the collection, we can work with that as a grapes Component object (see API Reference)
  let el = body.find("SomeQueryIdentifier"); // This will find the collection of elements that match the query and 
  //will also work as a Component object. Works only with already rendered components.
 
  // Then is just work with that element the way is needed and possible. 
  // For example, if you want do change some properties of it:
  el.setAttributes({ id: 'test', 'data-key': 'value' });

  // Or replace with some other element
  el.replaceWith("<div>Teste</div>")
```

And this is the easiest way to update or replace some element, including adding and removing scripts from the page.

**Reply by ronaldohoch** [reply]
Thank you Gustavo xD

For the next who try to do the same thing, here the working code of my sample:
https://jsfiddle.net/hke24d59/1/
